### PR TITLE
[FNDJEN-1482] - Use Jenkins base image to avoid workaround for Docker packaging limitations

### PR DESCRIPTION
**Pull request by oleg-nenashev**
This change works around limitations of Jenkins plugin management scripts from https://github.com/jenkinsci/docker which are not compatible with CloudBees Jenkins Distribution 2.176.2.3 structure. Instead of trying to hack the image, let's use CJD WAR file as a drop-in replacement.

Works around:

* https://github.com/jenkinsci/docker/issues/857
* https://github.com/jenkinsci/docker/issues/861

Once the issues are fixed, we can easily return to the CJD base image

### CODE REVIEW

**Review by MRamonLeon — APPROVED** [review]
This approach looks better to me.

File: Dockerfile [hunk 1]
```diff
@@ -1,31 +1,16 @@
...
-
-ENV JENKINS_UC http://jenkins-updates.cloudbees.com
-ENV JENKINS_INCREMENTALS_REPO_MIRROR https://repo.jenkins-ci.org/incrementals
-
-# Workaround https://github.com/jenkinsci/docker/issues/857
-USER root
-RUN mkdir -p /usr/share/jenkins/ && ln -s /usr/share/cloudbees-jenkins-distribution/cloudbees-jenkins-distribution.war /usr/share/jenkins/jenkins.war
-USER cloudbees-jenkins-distribution
+FROM jenkins/jenkins:2.176.2
+COPY --from=war_source /usr/share/cloudbees-jenkins-distribution/cloudbees-jenkins-distribution.war /usr/share/jenkins/jenkins.war
```
Comment (jphartley) [inline Dockerfile:6]: Only real comment is maybe add one line of comment here to explain to the user why you are doing this. This isn't super obvious to myself who is a relatively inexperienced Jenkins/CJD user.

**Review by jphartley — CHANGES_REQUESTED** [review]
Overall looks like a much cleaner approach, though I am not an expert on this. Shorter looks beter. Just maybe add a little more comments.

**Comment by oleg-nenashev** [comment]
@jphartley  done

**Review by varyvol — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (jphartley) [reply Dockerfile:6]: Thanks for the change

**Review by jphartley — APPROVED** [review]
Looks good now

**Pull request merged**